### ISSUE TITLE
Support integer-based enums

### ISSUE THREAD

**Issue by bartoszirzyk** [apple/swift-openapi-generator#241]
Hi, I've got an enum defined in yaml file:
```
            startMinute:
              type: object
              required:
              - value
              - type
              - unit
              properties:
                value:
                  type: integer
                  enum:
                  - 0
                  - 10
                  - 20
                  - 30
                  - 40
                  - 50
                  example: 10
                type:
                  type: string
                  example: number
                unit:
                  type: string
                  example: minute
```

this is parsed to:
```
public var value: Swift.Int
public var _type: Swift.String
public var unit: Swift.String
```

I expect that if it is an enum, the `value` should be an enum type with Int as it's rawValue

**Comment by czechboy0** [comment]
Hi @bartoszirzyk, thanks for reporting the issue.

You're right, we don't currently support integer-backed enums, only string-backed enums.

The JSON Schema specification allows them, however: https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-01#name-enum

**Comment by czechboy0** [comment]
@bartoszirzyk a fix for this is in #242, please try that branch and let me know if it fixes the issue for you. I'll look into landing the fix next week.

**Comment by czechboy0** [comment]
Ok it's landed to main now, please verify that it's fixed, before we tag a release.

**Comment by bartoszirzyk** [comment]
I’ll test that on Monday 🙌

**Comment by bartoszirzyk** [comment]
@czechboy0 It seems to work. Code is generated wit an enum. Thanks!

**Comment by czechboy0** [comment]
Released in 0.2.1
